pkg/config/templateComponent.go

```diff
@@ -185,6 +185,37 @@ func (t *TemplateComponent) Props() map[string]TemplateProperty {
 	return props
 }
 
+func (t *TemplateComponent) Data() map[string]any {
```

Comment:
I'm tempted to call this `Values` to mimic helm, but thats just bc that templating engine is what I'm used to.

Reply:
I like that better! i'm going to make that change.